### PR TITLE
Updating Gem versions - specifically updating Chef for omnibus

### PR DESCRIPTION
**Pull request by johnmccrae**
Signed-off-by: John <john.mccrae@progress.com>

Updating the chef version in Omnibus.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

**Review by tpowell-progress — APPROVED** [review]

**Comment by tpowell-progress** [comment]
🤔  it looks clean enough, but can we redo this without the big merge commit up front?

**Comment by neha-p6** [comment]
Do we an adhoc build testing this? If not, can we run an adhoc build please?

File: Gemfile.lock [hunk 1]
```diff
@@ -27,19 +27,6 @@ GIT
       http-cookie (>= 1.0.2, < 2.0)
       mime-types (>= 1.16, < 4.0)
       netrc (~> 0.8)
-    rest-client (2.1.0-x64-mingw-ucrt)
-      ffi (~> 1.15)
-      http-accept (>= 1.7.0, < 2.0)
-      http-cookie (>= 1.0.2, < 2.0)
-      mime-types (>= 1.16, < 4.0)
-      netrc (~> 0.8)
```
Comment (jaymzh) [inline Gemfile.lock:35]: This looks like we're dropping REST from Windows platform. I know that we have a bunch of changes non-responsive-upstream hasn't accepted, is that why? How is Chef on windows getting a REST client now?

**Review by jaymzh — CHANGES_REQUESTED** [review]
Can we get a description here of all the things we're changing and why? We spent about 10 minutes in the PR Review meeting digging around in the rest-client/rest-client PRs and this one and still can't quite piece this all together enough to give a meaningful review. Thanks!

File: Gemfile.lock (hunk 1 above)
Comment (johnmccrae) [reply Gemfile.lock]: That was a total gaffe. Updating the gemfile.lock on Windows is a total crap-shoot sometimes. Now fixed

**Review by Stromweld — APPROVED** [review]
LGTM

**Review by tpowell-progress — APPROVED** [review]
Ok, much cleaner. Let’s make sure to squash and merge at the end

File: Gemfile.lock [hunk 2]
```diff
@@ -518,4 +518,4 @@ DEPENDENCIES
   webmock
 
 BUNDLED WITH
-   2.3.18
+   2.3.17
```
Comment (PrajaktaPurohit) [inline Gemfile.lock:521]: should this be 2.3.18?

**Comment by PrajaktaPurohit** [comment]
Could we paste the passing adhoc builds before merging.

File: Gemfile.lock (hunk 2 above)
Comment (johnmccrae) [reply Gemfile.lock:521]: corrected

**Comment by johnmccrae** [comment]
Here's the relevant AdHoc build - everything passes except the enterprise linux pieces which are failing for everything right now
https://buildkite.com/chef/chef-chef-main-omnibus-adhoc/builds/1760

**Pull request merged**